lib/agcaldav/client.rb

```diff
@@ -190,13 +192,80 @@ def create_event event
       find_event(event.uid)
     end
 
-    def update_event event
-      #TODO... fix me
-      if delete_event event[:uid]
-        create_event event
-      else
-        return false
+    # Careful: etag needs surrounding quotes
+    def update_event(etag, event, data)
```

Comment:
Let's move this to the create_event method as discussed.

Maybe rename the method to `create_or_update` or `set_event` ...